code/del.dm

```diff
@@ -77,6 +77,8 @@ proc/qdel(var/datum/O)
 
 /datum/var/tmp/disposed = 0
 /datum/var/tmp/qdeled = 0
+/datum/var/tmp/thing_is_being_used = 1 // hi im using this, please don't steal it, thank you
+
```

Comment:
????

Reply:
There was an obnoxious problem with (what appeared to be) projectile pooling. Whenever the spell tried to create a projectile, half the time it'd unpool one of the missiles that had already been unpooled, making half the projectiles just not appear. I don't know why this happened, Corners mentioned something about how drones would sometimes steal homing vampire projectiles and that sounds like this, but after adding these, it stopped.

And I tested to make sure it doesnt just outright prevent things from being unpooled. I cast the spell multiple times and counted how many times it hit the "thing is in use, don't unpool it, make a new one" branch, and that number was far less than the number of projectiles that were made. So, this should also help issues where existing things get stolen by the pool!

As for why its a /tmp/ var, I dunno, the other vars were and I'm still not super sure what /tmp/ actually does, but it seems to work I guess?

Reply:
Please name the var better.
Tmp is for serialization.

Reply:
Turns out I didnt need that var, just setting the unpooler to check if the thing is in the pool looks like it works just fine.